src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -280,6 +280,13 @@ private static void MakeDeclarationNullable(SyntaxEditor editor, SemanticModel m
                 return propertyDeclarationSyntax.Type;
             }
 
+            // string x;
+            // Unassigned value that's not marked as null
+            if (node is VariableDeclaratorSyntax && node.Parent is VariableDeclarationSyntax variableDeclarationSyntax)
```

Comment:
Does this need to do a check that the VariableDeclarationSyntax is parented by a field, as this could potentially run elsewhere? Also, given a scenario like:

    string s, s2 = "hello"

This would mean we'd be switching both the declarations over to string? which probably isn't what the user wanted either.

Maybe "check if it's a field, and there's exactly one declarator" only then offer the fix?

Reply:
can also use pattern matching like `node is VariableDeclaratorSyntax { Parent: VariableDeclaration { Parent: FieldDeclarationSyntax _ } declaration } && declaration.Variables.Length == 1`.  Or something to that effect.